expertise/create_dataset.py

```diff
@@ -85,17 +87,59 @@ def deduplicate_publications(self, publications):
+            return None
+            
+        # Get domain from either domain field or invitation prefix
+        domain = getattr(pub, 'domain', None)
+        if domain is None:
+            domain = pub.invitation.split('/-/')[0]  # API1 fallback to invitation
+        
+        # Return early on DBLP papers (venueid =/= domain) and non-accepted papers (domain not in venue_list)
+        if not (venueid == domain and domain in self.venue_list):
+            return None
+        ## Papers allowed: accepted papers from an OpenReview venue
+            
+        # Find matching weight specification
+        fallback_weight = None
+        for venue_spec in weight_specification:
+            if 'prefix' in venue_spec and venueid.startswith(venue_spec['prefix']):
+                return venue_spec['weight']
+            if 'value' in venue_spec and venueid == venue_spec['value']:
+                return venue_spec['weight']
+            if 'inOpenReview' in venue_spec:
+                fallback_weight = venue_spec['weight']
+
+        return fallback_weight ## Prioritize venue match, then fallback to geenral OpenReview upweight
 
     def get_publications(self, author_id):
 
         dataset_params = self.config.get('dataset', {})
+        weight_specification = dataset_params.get('weight_specification', None)
         minimum_pub_date = dataset_params.get('minimum_pub_date') or dataset_params.get('or', {}).get('minimum_pub_date', 0)
         top_recent_pubs = dataset_params.get('top_recent_pubs') or dataset_params.get('or', {}).get('top_recent_pubs', False)
-
         publications = self.deduplicate_publications(
             self.get_paper_notes(author_id, dataset_params)
         )
 
+        # Validate weight specification
+        if weight_specification:
+            if not isinstance(weight_specification, list):
+                raise ValueError('weight_specification must be a list')
+            for venue_spec in weight_specification:
+                if not isinstance(venue_spec, dict):
+                    raise ValueError('Objects in weight_specification must be dictionaries')
+                if 'prefix' in venue_spec and 'value' in venue_spec and 'inOpenReview' in venue_spec:
+                    raise KeyError('Objects in weight_specification must only have one of [prefix, value, inOpenReview]')
+                if 'prefix' not in venue_spec and 'value' not in venue_spec and 'inOpenReview' not in venue_spec:
+                    raise KeyError('Objects in weight_specification must have a prefix, value, or inOpenReview key')
+                if 'weight' not in venue_spec:
+                    raise KeyError('Objects in weight_specification must have a weight key')
+                # weight must be an integer or float
+                if not isinstance(venue_spec['weight'], int) and not isinstance(venue_spec['weight'], float):
+                    raise ValueError('weight must be an integer')
```

Comment:
I think the following following should be:
`raise ValueError('weight must be a number')`
If we are only allowing positive numbers:
`raise ValueError('weight must be a number greater than 0')`

Reply:
I do not see these changes in the code, why is this conversation resolved?